test/integration/selector_test.go

```diff
@@ -0,0 +1,200 @@
+package integration
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSelector(t *testing.T) {
+	sysCfg := baseSysCfg(t)
+	initTest(t, sysCfg)
+
+	const (
+		commonLabelKey     = "test-label-key"
+		m0m1CommonLabelVal = "test-label-val-found"
+		m2CommonLabelVal   = "test-label-val-not-found"
+
+		m0OnlyLabelKey = "m0-only-label-key"
+		m0OnlyLabelVal = "m0-only-label-val"
+		m1OnlyLabelKey = "m1-only-label-key"
+		m1OnlyLabelVal = "m1-only-label-val"
+	)
+	// Model with an active backend to send requests to.
+	m0 := modelForTest(t)
+	m0.Name = m0.Name + "0"
+	m0.Labels[commonLabelKey] = m0m1CommonLabelVal
+	m0.Labels[m0OnlyLabelKey] = m0OnlyLabelVal
+	require.NoError(t, testK8sClient.Create(testCtx, m0))
+
+	testModelBackend := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		log.Println("Serving request from testBackend")
+		w.WriteHeader(200)
+	}))
+	updateModelWithBackend(t, m0, testModelBackend)
+
+	// Update with MinReplicas after setting annotation to trigger Pod creation
+	// with annotations pointing to test backend.
+	updateModel(t, m0, func() {
+		m0.Spec.MinReplicas = 1
+	}, "Set MinReplicas to 1")
+	requireModelPods(t, m0, 1, "Min replica Pod should be created", 5*time.Second)
+	markAllModelPodsReady(t, m0)
+
+	//logPods(t)
+
+	inferenceCases := []struct {
+		name            string
```

Comment:
personal preference: when you use a `map[string]struct{}` instead of a slice, you can use the name as key. I find that more readable.

Reply:
I like that, never thought to do it that way, changed.